### PR TITLE
feat: ✨ Delete Lobbies with Cron

### LINKED ISSUE #131: delete old lobbies from firestore
if all the players in a lobby decide to just close the browser at once then the lobby should be deleted eventually instead of taking up space in firestore forever because we don't want to be storing and possibly paying for a ton of old lobbies

Tasks:
- [ ] make expiration non-optional
- [ ] make cron job

Acceptance Criteria:
- lobbies should not be automatically deleted if they are less than a day after the game ends
- lobbies should be automatically deleted within a week of when the game ends
- hackers should not be able to prematurely delete a lobby

### PR DESCRIPTION
**Pull request by gchamb**
How to Test:

- Create a Lobby
- Change the expiration in the lobby doc to 24 hours ago or more
- Click the delete button on the splash screen page (Only there for testing)

### CODE REVIEW

**Review by Muaad2001 — DISMISSED** [review]
checks out

File: functions/src/lobby.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ export const createLobby = functions.https.onCall(async (data: unknown, context)
     ],
     state: "WAIT",
     alivePlayers: [context.auth.uid],
+    expiration: firestore.Timestamp.now(),
```
Comment (nstringham) [inline functions/src/lobby.ts]: maybe set this to like 2 or 3 hours instead of instant

Comment (nstringham) [reply functions/src/lobby.ts]: don't forget to also do this when the game ends and the lobby goes back to WAIT

File: functions/src/cron.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export const testLogic = functions.https.onRequest(async (req, res) => {
+  deleteOldLobbies();
+  res.send({ message: "Hi" });
+});
+
+// Schedule for every day at 3:15AM
+// export const deleteOldLobbies = functions.pubsub.schedule("15 3 * * *").onRun(async (context) => {
+const deleteOldLobbies = async () => {
+  // TODO: Remove if not testing the logic
+  const dayFromTodaySecs = firestore.Timestamp.now().seconds - 86_400; // A day ago from today in seconds
```
Comment (nstringham) [inline functions/src/cron.ts]: I would prefer if this was `24 * 60 * 60`

File: functions/src/cron.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  const dayFromTodayDate = firestore.Timestamp.fromDate(new Date(dayFromTodaySecs * 1000)); // a day ago from today in date object
```
Comment (nstringham) [inline functions/src/cron.ts]: let's call these constants something that if obvious to everone that it refers to the past. maybe we could call it `oneDayAgo` or `twentyFourHoursAgo`

**Review by nstringham — DISMISSED** [review]
looks pretty good don't forget to remove the testing things

**Review by mrjoshua520 — APPROVED** [review]

**Review by nstringham — APPROVED** [review]

**Pull request merged**